### DISCUSSION TITLE
❓4G connectivity

### DISCUSSION THREAD

**Discussion by no-name-99** [isontheline/pro.webssh.net#476] [General]
It seems that the app can not connect over 4G at all. Any suggestions please ?

**Comment by isontheline** [comment]
Hello @no-name-99 👋

Could you post the error thrown?

Have a nice day☀️

**Reply by zombiRon** [reply]
I have this issue with some hosts, but others work fine. All hosts work when using wifi, but switch to cellular and it's hit and miss. I wondered if it was my cell provider having a routing issue, but I managed to connect with another app (termius). The error is "connection failed, can not establish connection to xx.xx.xx.xx"

**Reply by isontheline** [reply]
Does the host is IPv6?

When the error occurs, could you try to ping from inside WebSSH ping tool?

**Reply by zombiRon** [reply]
Yeah good shout. The host is ip4, but ping (tools) replies with an ip6 address, and ping (mashREPL) says "request timeout". 

You have different methods in your app??

**Reply by isontheline** [reply]
ping (tools) is IPv6 aware but ping (mashREPL) isn't so it complains with no IPv4 destination available.

Are you using a domain name which returns a different IPv4 / IPv6 when "devices are IPv6 aware" (DNS)?

For example webssh.net returns an IPv4 or (if you are on a IPv6 network) IPv6.

You can also try to tell WebSSH to prefer IPv4 :
<img width="1190" alt="image" src="https://user-images.githubusercontent.com/44212923/182457802-dd9360d8-83d6-48d4-b5fe-e21feee2ced2.png">

**Reply by zombiRon** [reply]
Host was defined in webssh config as just an ip4 address. 

I have done a reverse lookup on the address and used the resulting record as the host, webssh now connects over cellular. 

Many thanks for your help, can this be added to the documentation to help future users??

**Reply by isontheline** [reply]
> Many thanks for your help

You're Welcome

> Can this be added to the documentation to help future users??

Could you tell me more about "I have done a reverse lookup on the address and used the resulting record as the host"?
